xarray/core/dataarray.py

```diff
@@ -1085,6 +1085,60 @@ def to_masked_array(self, copy=True):
         isnull = pd.isnull(self.values)
         return np.ma.MaskedArray(data=self.values, mask=isnull, copy=copy)
 
+    def to_dict(self):
+        """Convert this xarray.DataArray into a dictionary following xarray naming conventions.
+        """
+        d = {'coords': {}, 'attrs': dict(self.attrs), 'dims': self.dims}
+
+        def time_check(val):
+            if np.issubdtype(val.dtype, np.datetime64):
+                val = val.astype('datetime64[us]')
+            elif np.issubdtype(val.dtype, np.timedelta64):
+                val = val.astype('timedelta64[us]')
+            return(val)
+
+        for k in self.coords:
+            d['coords'].update({k: {'data': time_check(self[k].values).tolist(),
+                                    'dims': self[k].dims,
+                                    'attrs': dict(self[k].attrs)}})
+
+        d.update({'data': time_check(self.values).tolist(),
+                  'name': self.name})
+        return d
+
+    @classmethod
+    def from_dict(cls, d):
+        """
+        Convert a dictionary into an xarray.DataArray
+
+        Parameters
+        ----------
+        d : dict, with a minimum structure of {'dims': [..], 'data': [..]}
+            or optionally {'dims': [..], 
+                           'data': [..], 
+                           'coords': {'dims': [..], 'data': [..]}}
+
+        Returns
+        -------
+        obj : xarray.DataArray
+
+        """
+        coords = None
+        if 'coords' in d.keys():
```

Comment:
You don't need `.keys()` here -- you can check dictionaries directly. Generally that's considered better style, especially because it can be very slow to check `.keys()` on Python 2 (because it's converted into a list).